src/RequestProcessor.cs

```diff
@@ -24,33 +25,61 @@ internal class RequestProcessor
+
+            // Host sends terminate message to worker.
+            // Worker terminates if it can, otherwise host terminates after a grace period
+            _requestHandlers.Add(StreamingMessage.ContentOneofCase.WorkerTerminate, ProcessWorkerTerminateRequest);
+
+            // Add any worker relevant status to response
+            _requestHandlers.Add(StreamingMessage.ContentOneofCase.WorkerStatusRequest, ProcessWorkerStatusRequest);
+
+            // On file change event, host sends notification to worker
+            _requestHandlers.Add(StreamingMessage.ContentOneofCase.FileChangeEventRequest, ProcessFileChangeEventRequest);
+
+            // Host sends required metadata to worker to load function
+            _requestHandlers.Add(StreamingMessage.ContentOneofCase.FunctionLoadRequest, ProcessFunctionLoadRequest);
+            
+            // Host requests a given invocation
+            _requestHandlers.Add(StreamingMessage.ContentOneofCase.InvocationRequest, ProcessInvocationRequest);
+
+            // Host sends cancel message to attempt to cancel an invocation. 
+            // If an invocation is cancelled, host will receive an invocation response with status cancelled.
+            _requestHandlers.Add(StreamingMessage.ContentOneofCase.InvocationCancel, ProcessInvocationCancelRequest);
+
+            _requestHandlers.Add(StreamingMessage.ContentOneofCase.FunctionEnvironmentReloadRequest, ProcessFunctionEnvironmentReloadRequest);
         }
 
         internal async Task ProcessRequestLoop()
         {
+            var logger = new RpcLogger(_msgStream);
+
             StreamingMessage request, response;
             while (await _msgStream.MoveNext())
             {
                 request = _msgStream.GetCurrentMessage();
-                switch (request.ContentCase)
+
+                if (_requestHandlers.TryGetValue(request.ContentCase, out Func<StreamingMessage, StreamingMessage> requestFunc))
+                {
+                    response = requestFunc(request);
+                }
+                else
                 {
-                    case StreamingMessage.ContentOneofCase.WorkerInitRequest:
-                        response = ProcessWorkerInitRequest(request);
-                        break;
-                    case StreamingMessage.ContentOneofCase.FunctionLoadRequest:
-                        response = ProcessFunctionLoadRequest(request);
-                        break;
-                    case StreamingMessage.ContentOneofCase.InvocationRequest:
-                        response = ProcessInvocationRequest(request);
-                        break;
-                    default:
-                        string errorMsg = string.Format(PowerShellWorkerStrings.UnsupportedMessage, request.ContentCase);
-                        throw new InvalidOperationException(errorMsg);
+                    logger.SetContext(request.RequestId, null);
```

Comment:
Do you think `SetContext/ResetContext` can be skipped here? `StreamingMessage.RequestId` is null (or an empty string, cannot remember clearly) all the time ...